server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -629,6 +630,7 @@ boolean isIncomplete() {
 
             stopperRunner.accept("http-server-transport-stop", injector.getInstance(HttpServerTransport.class)::close);
             stopperRunner.accept("async-search-stop", () -> awaitSearchTasksComplete(maxTimeout));
+            stopperRunner.accept("reindex-stop", () -> awaitReindexTasksComplete(TimeValue.min(TimeValue.timeValueMinutes(1), maxTimeout)));
```

Comment:
Let us make the timeout configurable separately from the max-timeout. I think 1 min could be too long by default, I'd go with something like 10secs and we can increase where we need it. The problem is that users do rolling restarts and this could prevent those from completing as fast as before - and the longer it takes to restart the more operations may need to be replayed to re-establish sync'ed replicas.